server.js

```diff
@@ -1,39 +1,117 @@
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
-
-const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo"
-mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
-mongoose.Promise = Promise
-
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
-const port = process.env.PORT || 8080
-const app = express()
+import express from "express";
+import bodyParser from "body-parser";
+import cors from "cors";
+import mongoose from "mongoose";
+
+import songs from "./data/top-music.json";
+import genres from "./data/music-genres.json";
+
+const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo";
+mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true });
+mongoose.Promise = Promise;
+
+const songSchema = new mongoose.Schema({
+  id: Number,
+  trackName: String,
+  artistName: String,
+  genre: {
+    type: mongoose.Schema.Types.ObjectId,
+    ref: "Genre",
+  },
+  bpm: Number,
+  energy: Number,
+  danceability: Number,
+  loudness: Number,
+  liveness: Number,
+  valence: Number,
+  length: Number,
+  acousticness: Number,
+  speechiness: Number,
+  popularity: Number,
+});
+
+const Song = mongoose.model("Song", songSchema);
+
+const genreSchema = new mongoose.Schema({
+  description: String,
+});
+
+const Genre = mongoose.model("Genre", genreSchema);
+
+if (process.env.RESET_DB) {
+  const seedDB = async () => {
+    await Song.deleteMany();
+    await Genre.deleteMany();
+
+    const genresArray = [];
+
+    genres.forEach(async (item) => {
+      const genre = new Genre(item);
+      genresArray.push(genre);
+      await genre.save();
+    });
+
+    songs.forEach(async (item) => {
+      const newSong = new Song({
+        ...item,
+        genre: genresArray.find(
+          (singleGenre) => singleGenre.description === item.genre
+        ),
+      });
+      await newSong.save();
+    });
+  };
+
+  seedDB();
+}
+
+const port = process.env.PORT || 8080;
+const app = express();
 
 // Add middlewares to enable cors and json body parsing
-app.use(cors())
-app.use(bodyParser.json())
+app.use(cors());
+app.use(bodyParser.json());
 
 // Start defining your routes here
-app.get('/', (req, res) => {
-  res.send('Hello world')
-})
+app.get("/", (req, res) => {
+  res.send("Hello world");
+});
+
+app.get("/songs", async (req, res) => {
+  const { song_name } = req.query;
+
+  if (song_name) {
+    const songs = await Song.find({
+      trackName: {
+        $regex: new RegExp(song_name, "i"),
+      },
+    });
+    res.json(songs);
+  } else {
+    const songs = await Song.find();
+    res.json(songs);
+  }
+});
+
+app.get("/songs/:songId", async (req, res) => {
+  const { songId } = req.params;
+  const singleSong = await Song.findOne({ _id: songId });
```

Comment:
When you are using _id field you have a nice variety of very fast functions on hand:
https://mongoosejs.com/docs/api.html#model_Model.findById